pkg/channel/producer/producer.go

```diff
@@ -108,6 +117,20 @@ func (p *Producer) ProduceKafkaMessage(ctx context.Context, channelReference eve
 	}
 }
 
+// Returns A CloudEvents TransformerFunc For Setting The "partitionkey" Extension To The Message's Subject If Not Already Set
+func DefaultPartitionKeyTransformer() binding.TransformerFunc {
```

Comment:
Are you mutating the event here right? In theory, you should not. A tip: if you need to set the partition key from the subject, a simple solution could be to:

Create a closure as transformer (using `TransformerFunc`) that changes a local variable, like "wannaBeKey":

```
var partitionKey string
keyExtractor := binding.TransformerFunc(func(reader, writer) {
  partitionKey = .... 
})
```

then pass this transformer when you invoke `WriteProducerMessage`. At that point the transformer was invoked and you should see the key (if any) in `partitionKey` variable. Now you can assign it to `message.Key`. If you don't like the approach, you can always enclose in this transformer a string pointer, so you write the variable from that pointer. Just look at the code inside `WriteProducerMessage` and you see i do a similar approach.

Reply:
As per the comment below I will be removing this as well.  If we need this in our application we can do it further upstream (outside Knative paradigms) which is cleaner - thanks!